//
// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package zetasql.functions;

import "zetasql/public/type.proto";

option java_package = "com.google.zetasql.functions";
option java_outer_classname = "ZetaSQLBitwiseAggMode";

// Enum that defines the aggregation behavior of the BIT_AND, BIT_OR, and
// BIT_XOR functions when there are multiple BYTES values of different lengths.
// See (broken link).
message BitwiseAggEnums {
  enum BitwiseAggMode {
    option (opaque_enum_type_options).sql_opaque_enum_name = "BITWISE_AGG_MODE";

    // Invalid in ZetaSQL.
    BITWISE_AGG_MODE_INVALID = 0
        [(opaque_enum_value_options).invalid_enum_value = true];

    // If the length of any of the BYTES arguments is different from the others,
    // throw an error.
    STRICT = 1;

    // Right pad (left align) all BYTES values shorter than the longest one with
    // b'\x00'
    PAD = 2;
  }
}
